### PR TITLE
CB-16169 Create default template API in datalake

### PR DESCRIPTION
**Pull request by lturcsanyi**
CB-16169 Create default template API in datalake

### CODE REVIEW

File: datalake-api/src/main/java/com/sequenceiq/sdx/api/endpoint/SdxEndpoint.java [hunk 1]
```diff
@@ -289,4 +290,17 @@ ObjectStorageValidateResponse validateCloudStorage(@ValidStackNameFormat @ValidS
...
+
+    @GET
+    @Path("recommendation")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Creates a recommendation that advises cloud resources for the given datalake shape based on the given credential crn.",
+            nickname = "getRecommendationByCredCrn")
+    SdxRecommendationResponse getRecommendationByCredCrn(
+            @QueryParam("credentialCrn") String credentialCrn,
+            @QueryParam("region") String region,
+            @QueryParam("platformVariant") String platformVariant,
```
Comment (lnardai) [inline datalake-api/src/main/java/com/sequenceiq/sdx/api/endpoint/SdxEndpoint.java:302]: Platform variant could be inferred from credentialCrn I think.

File: datalake/src/main/java/com/sequenceiq/datalake/configuration/AppConfig.java [hunk 2]
```diff
@@ -51,6 +57,15 @@
     @Named("freeIpaServerUrl")
     private String freeIpaServerUrl;
 
+    @Value("${datalake.intermediate.threadpool.core.size}")
```
Comment (lnardai) [inline datalake/src/main/java/com/sequenceiq/datalake/configuration/AppConfig.java:60]: Why is this needed?

File: datalake/src/main/java/com/sequenceiq/datalake/service/cloud/CloudResourceRetrieverService.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import com.sequenceiq.cloudbreak.cloud.service.ResourceRetriever;
+import com.sequenceiq.common.api.type.CommonStatus;
+import com.sequenceiq.common.api.type.ResourceType;
+
+@Component
+public class CloudResourceRetrieverService implements ResourceRetriever {
+
+    @Override
+    public Optional<CloudResource> findByResourceReferenceAndStatusAndType(String resourceReference, CommonStatus status, ResourceType resourceType) {
+        return Optional.empty();
```
Comment (lnardai) [inline datalake/src/main/java/com/sequenceiq/datalake/service/cloud/CloudResourceRetrieverService.java:17]: Looks like a missing implementation.

File: datalake/src/main/java/com/sequenceiq/datalake/configuration/AppConfig.java (hunk 2 above)
Comment (lturcsanyi) [reply datalake/src/main/java/com/sequenceiq/datalake/configuration/AppConfig.java:60]: It is necessary because of the added cloud-xxx dependencies.

File: datalake/src/main/java/com/sequenceiq/datalake/service/cloud/CloudResourceRetrieverService.java (hunk 3 above)
Comment (lturcsanyi) [reply datalake/src/main/java/com/sequenceiq/datalake/service/cloud/CloudResourceRetrieverService.java:17]: Copied from redbeams service, this is also because of the added dependencies.

**Comment by aszegedi** [comment]
retest this please

**Comment by horadla23** [comment]
as discussed, let's check if it is an option to reuse existing recommandation CB API on UI for this and include only defaultTemplate in the new DL API

**Review by horadla23 — APPROVED** [review]
